src/common/setup.ts

```diff
@@ -0,0 +1,179 @@
+import { existsSync, readdirSync } from 'fs'
+import { load } from 'js-yaml'
+import { fileURLToPath } from 'url'
+import { $, nothrow } from 'zx'
+import { decrypt } from './crypt'
+import { OtomiDebugger } from './debug'
+import { values } from './hf'
+import { BasicArguments, ENV, parser } from './no-deps'
+import { evaluateSecrets } from './secrets'
+import { askYesNo, source } from './zx-enhance'
+
+const dirname = fileURLToPath(import.meta.url)
+
+let otomiImageTag: string
+let otomiCustomerName: string
+let otomiK8sVersion: string
+
+/**
+ * Check whether the environment matches the configuration for the Kubernetes Context
+ * @param debug
+ * @returns
+ */
+const checkKubeContext = async (debug: OtomiDebugger): Promise<void> => {
+  if (ENV.isCI) return
+  debug.verbose('Validating kube context')
+
+  const envPath = `${ENV.DIR}/env/.env`
+  try {
+    await source(envPath)
+  } catch (error) {
+    debug.exit(1, error)
+  }
+
+  if (!('K8S_CONTEXT' in process.env)) debug.exit(1, `K8S_CONTEXT is not defined in '${envPath}'`)
+  debug.verbose(`Using kube context: ${process.env.K8S_CONTEXT}`)
+
+  // TODO: Consider using the kubernetes-client: https://github.com/kubernetes-client/javascript
+  const runningContext = (await nothrow($`kubectl config current-context`)).stdout.trim()
+  if (process.env.K8S_CONTEXT !== runningContext) {
+    let fixContext = false
+    if (!(parser.argv as BasicArguments).setContext) {
+      fixContext = await askYesNo(
+        `Warning: Your current kubernetes context does not match cluster context: ${process.env.K8S_CONTEXT}. Would you like to switch kube context to cluster first?`,
+        { defaultYes: true },
+      )
+    }
+    if (fixContext || (parser.argv as BasicArguments).setContext) {
+      await $`kubectl config use ${process.env.K8S_CONTEXT}`
+    }
+  }
+}
+/**
+ * Check the ENV_DIR parameter and whether or not the folder is populated
+ * @param debug
+ * @returns
+ */
+const checkENVdir = (debug: OtomiDebugger): boolean => {
+  if (dirname.includes('otomi-core') && !('ENV_DIR' in process.env)) {
+    debug.exit(1, 'The ENV_DIR environment variable is not set')
```

Comment:
What do you think about a prompt to set an initial ENV_DIR, e.g. one of the profiles?

Reply:
Hmm interesting, but I don't think we should go that route. Since we use that to bootstrap, if we use the profile path, then we overwrite default bootstrap values. Not sure if that is useful?

Reply:
Not doing this as profiles no longer exist.